Windows doesn't build since GCC 11 upgrade

@peterbud @houz @wpferguson Seems to be due to [```#undef __STRICT_ANSI__```](https://github.com/darktable-org/darktable/blob/f6bdd28abdb19808a4b15fe9d09cca72e1b6c5e8/src/win/win.h#L8)

See e.g. log of the first failure of the nightly build when mingw gcc was upgraded from 10.3 to 11.2: https://github.com/darktable-org/darktable/actions/workflows/nightly.yml

```
[ 19%] Building CXX object bin/CMakeFiles/lib_darktable.dir/common/exif.cc.obj
cd /D/a/darktable/darktable/build/bin && /D/a/_temp/msys64/mingw64/bin/g++.exe -DAVIF_DLL -DGDK_DISABLE_DEPRECATED -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_22 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_MIN_REQUIRED -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_56 -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -DHAVE_BUILTIN_CPU_SUPPORTS -DHAVE_CONFIG_H -DHAVE_GAME -DHAVE_GMIC -DHAVE_GPHOTO2 -DHAVE_GPHOTO_25_OR_NEWER -DHAVE_GRAPHICSMAGICK -DHAVE_HTTP_SERVER -DHAVE_ICU -DHAVE_ISO_CODES -DHAVE_LENSFUN -DHAVE_LIBAVIF=1 -DHAVE_LIBEXIV2_WITH_ISOBMFF=1 -DHAVE_LIBHEIF=1 -DHAVE_LIBSECRET -DHAVE_MAP -DHAVE_OPENCL -DHAVE_OPENEXR -DHAVE_OPENJPEG -DHAVE_OSMGPSMAP_110_OR_NEWER -DHAVE_SQLITE_324_OR_NEWER -DSQLITE_CORE -DSQLITE_ENABLE_ICU -DUSE_LUA -D_POSIX_THREAD_SAFE_FUNCTIONS -D__GDK_KEYSYMS_COMPAT_H__ -D__USE_MINGW_ANSI_STDIO=1 -Dlib_darktable_EXPORTS -I/D/a/darktable/darktable/build/bin -I/D/a/darktable/darktable/src/src -I/D/a/darktable/darktable/src/src/external/LuaAutoC -I/D/a/darktable/darktable/src/src/external/whereami/src -isystem /D/a/darktable/darktable/src/src/external -isystem /D/a/darktable/darktable/src/src/external/OpenCL -isystem /D/a/_temp/msys64/mingw64/include/glib-2.0 -isystem /D/a/_temp/msys64/mingw64/lib/glib-2.0/include -isystem /D/a/_temp/msys64/mingw64/include/gtk-3.0 -isystem /D/a/_temp/msys64/mingw64/include/pango-1.0 -isystem /D/a/_temp/msys64/mingw64/include/harfbuzz -isystem /D/a/_temp/msys64/mingw64/include/freetype2 -isystem /D/a/_temp/msys64/mingw64/include/libpng16 -isystem /D/a/_temp/msys64/mingw64/include/fribidi -isystem /D/a/_temp/msys64/mingw64/include/cairo -isystem /D/a/_temp/msys64/mingw64/include/lzo -isystem /D/a/_temp/msys64/mingw64/include/pixman-1 -isystem /D/a/_temp/msys64/mingw64/include/gdk-pixbuf-2.0 -isystem /D/a/_temp/msys64/mingw64/include/atk-1.0 -isystem /D/a/_temp/msys64/mingw64/include/libxml2 -isystem /D/a/_temp/msys64/mingw64/include/libsoup-2.4 -isystem /D/a/_temp/msys64/mingw64/include/OpenEXR -isystem /D/a/_temp/msys64/mingw64/include/lensfun -isystem /D/a/_temp/msys64/mingw64/include/librsvg-2.0 -isystem /D/a/_temp/msys64/mingw64/include/json-glib-1.0 -isystem /D/a/_temp/msys64/mingw64/include/openjpeg-2.4 -isystem /D/a/_temp/msys64/mingw64/include/libsecret-1 -isystem /D/a/_temp/msys64/mingw64/include/GraphicsMagick -isystem /D/a/_temp/msys64/mingw64/include/osmgpsmap-1.0 -Wall -Wno-format -Wshadow -Wtype-limits -Wvla -Wno-unknown-pragmas -Wno-error=varargs -Wno-format-truncation -Wno-error=address-of-packed-member -std=c++14 -fopenmp -mtune=generic -msse2 -g -mfpmath=sse -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only -fexpensive-optimizations   -DUNICODE -D_UNICODE -Werror -Wfatal-errors -std=c++14 -MD -MT bin/CMakeFiles/lib_darktable.dir/common/exif.cc.obj -MF CMakeFiles/lib_darktable.dir/common/exif.cc.obj.d -o CMakeFiles/lib_darktable.dir/common/exif.cc.obj -c /D/a/darktable/darktable/src/src/common/exif.cc
In file included from D:/a/_temp/msys64/mingw64/include/c++/11.2.0/cstdlib:41,
                 from D:/a/_temp/msys64/mingw64/include/c++/11.2.0/stdlib.h:36,
                 from D:/a/_temp/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/11.2.0/include/mm_malloc.h:27,
                 from D:/a/_temp/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/11.2.0/include/xmmintrin.h:34,
                 from D:/a/_temp/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/11.2.0/include/immintrin.h:31,
                 from D:/a/_temp/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/11.2.0/include/x86intrin.h:32,
                 from D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/winnt.h:1555,
                 from D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/minwindef.h:163,
                 from D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/windef.h:9,
                 from D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from D:/a/_temp/msys64/mingw64/x86_64-w64-mingw32/include/winsock2.h:23,
                 from D:/a/darktable/darktable/src/src/win/win.h:13,
                 from D:/a/darktable/darktable/src/src/common/darktable.h:33,
                 from D:/a/darktable/darktable/src/src/control/control.h:21,
                 from D:/a/darktable/darktable/src/src/common/exif.cc:35:
D:/a/_temp/msys64/mingw64/include/c++/11.2.0/x86_64-w64-mingw32/bits/c++config.h:573:2: error: #warning "__STRICT_ANSI__ seems to have been undefined; this is not supported" [-Werror=cpp]
  573 | #warning "__STRICT_ANSI__ seems to have been undefined; this is not supported"
      |  ^~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus.exe: all warnings being treated as errors
make[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:404: bin/CMakeFiles/lib_darktable.dir/common/exif.cc.obj] Error 1
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -498,14 +498,16 @@ endif(BUILD_TESTING)
 # 	set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-m${USE_SSE_SET}")
 # endif(USE_SSE_SET)
 
-# we need some specific functions:
+# we need some specific functions/macros outside of ANSI C:
 if(NOT WIN32)
     if(CMAKE_SYSTEM MATCHES "SunOS.*")
         add_definitions("-D_XOPEN_SOURCE=600")
     else()
         add_definitions("-D_XOPEN_SOURCE=700")
     endif(CMAKE_SYSTEM MATCHES "SunOS.*")
-endif(NOT WIN32)
+else()
+    add_definitions("-D_USE_MATH_DEFINES")
+endif()
 
 # Set default component name - that way external modules like RawSpeed will install their
 # materials under the default component and not under 'Unspecified'

--- a/src/common/imageio_dng.h
+++ b/src/common/imageio_dng.h
@@ -207,12 +207,12 @@ static inline void dt_imageio_write_dng(
     const uint8_t xtrans[6][6],
     const float whitelevel,
     const dt_aligned_pixel_t wb_coeffs,
-    const char camera_model[24])
+    const char camera_makermodel[128])
 {
   FILE *f = g_fopen(filename, "wb");
   if(f)
   {
-    dt_imageio_dng_write_tiff_header(f, wd, ht, 1.0f / 100.0f, 1.0f / 4.0f, 50.0f, 100.0f, filter, xtrans, whitelevel, wb_coeffs, camera_model);
+    dt_imageio_dng_write_tiff_header(f, wd, ht, 1.0f / 100.0f, 1.0f / 4.0f, 50.0f, 100.0f, filter, xtrans, whitelevel, wb_coeffs, camera_makermodel);
     const int k = fwrite(pixel, sizeof(float), (size_t)wd * ht, f);
     if(k != wd * ht) fprintf(stderr, "[dng_write] Error writing image data to %s\n", filename);
     fclose(f);

--- a/src/iop/colorbalance.c
+++ b/src/iop/colorbalance.c
@@ -1407,6 +1407,10 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
 
   d->mode = p->mode;
 
+  const dt_aligned_pixel_t lift = { p->lift[CHANNEL_RED], p->lift[CHANNEL_GREEN], p->lift[CHANNEL_BLUE] };
+  const dt_aligned_pixel_t gamma = { p->gamma[CHANNEL_RED], p->gamma[CHANNEL_GREEN], p->gamma[CHANNEL_BLUE] };
+  const dt_aligned_pixel_t gain = { p->gain[CHANNEL_RED], p->gain[CHANNEL_GREEN], p->gain[CHANNEL_BLUE] };
+
   switch(d->mode)
   {
     case SLOPE_OFFSET_POWER:
@@ -1414,19 +1418,19 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
       // Correct the luminance in RGB parameters so we don't affect it
       dt_aligned_pixel_t XYZ;
 
-      dt_prophotorgb_to_XYZ((const float *)&p->lift[CHANNEL_RED], XYZ);
+      dt_prophotorgb_to_XYZ(lift, XYZ);
       d->lift[CHANNEL_FACTOR] = p->lift[CHANNEL_FACTOR];
       d->lift[CHANNEL_RED] = (p->lift[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->lift[CHANNEL_GREEN] = (p->lift[CHANNEL_GREEN] - XYZ[1]) + 1.f;
       d->lift[CHANNEL_BLUE] = (p->lift[CHANNEL_BLUE] - XYZ[1]) + 1.f;
 
-      dt_prophotorgb_to_XYZ((const float *)&p->gamma[CHANNEL_RED], XYZ);
+      dt_prophotorgb_to_XYZ(gamma, XYZ);
       d->gamma[CHANNEL_FACTOR] = p->gamma[CHANNEL_FACTOR];
       d->gamma[CHANNEL_RED] = (p->gamma[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->gamma[CHANNEL_GREEN] = (p->gamma[CHANNEL_GREEN] - XYZ[1]) + 1.f;
       d->gamma[CHANNEL_BLUE] = (p->gamma[CHANNEL_BLUE] - XYZ[1]) + 1.f;
 
-      dt_prophotorgb_to_XYZ((const float *)&p->gain[CHANNEL_RED], XYZ);
+      dt_prophotorgb_to_XYZ(gain, XYZ);
       d->gain[CHANNEL_FACTOR] = p->gain[CHANNEL_FACTOR];
       d->gain[CHANNEL_RED] = (p->gain[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->gain[CHANNEL_GREEN] = (p->gain[CHANNEL_GREEN] - XYZ[1]) + 1.f;
@@ -1452,19 +1456,20 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
     {
       // Correct the luminance in RGB parameters so we don't affect it
       dt_aligned_pixel_t XYZ;
-      dt_prophotorgb_to_XYZ((const float *)&p->lift[CHANNEL_RED], XYZ);
+
+      dt_prophotorgb_to_XYZ(lift, XYZ);
       d->lift[CHANNEL_FACTOR] = p->lift[CHANNEL_FACTOR];
       d->lift[CHANNEL_RED] = (p->lift[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->lift[CHANNEL_GREEN] = (p->lift[CHANNEL_GREEN] - XYZ[1]) + 1.f;
       d->lift[CHANNEL_BLUE] = (p->lift[CHANNEL_BLUE] - XYZ[1]) + 1.f;
 
-      dt_prophotorgb_to_XYZ((const float *)&p->gamma[CHANNEL_RED], XYZ);
+      dt_prophotorgb_to_XYZ(gamma, XYZ);
       d->gamma[CHANNEL_FACTOR] = p->gamma[CHANNEL_FACTOR];
       d->gamma[CHANNEL_RED] = (p->gamma[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->gamma[CHANNEL_GREEN] = (p->gamma[CHANNEL_GREEN] - XYZ[1]) + 1.f;
       d->gamma[CHANNEL_BLUE] = (p->gamma[CHANNEL_BLUE] - XYZ[1]) + 1.f;
 
-      dt_prophotorgb_to_XYZ((const float *)&p->gain[CHANNEL_RED], XYZ);
+      dt_prophotorgb_to_XYZ(gain, XYZ);
       d->gain[CHANNEL_FACTOR] = p->gain[CHANNEL_FACTOR];
       d->gain[CHANNEL_RED] = (p->gain[CHANNEL_RED] - XYZ[1]) + 1.f;
       d->gain[CHANNEL_GREEN] = (p->gain[CHANNEL_GREEN] - XYZ[1]) + 1.f;

--- a/src/win/win.h
+++ b/src/win/win.h
@@ -5,7 +5,6 @@
 #endif
 #define __MSVCRT_VERSION__ 0x0700
 
-#undef __STRICT_ANSI__
 #define XMD_H
 
 #undef _WIN32_WINNT

